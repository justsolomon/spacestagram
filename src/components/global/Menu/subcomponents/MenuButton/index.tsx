import styles from "./menuButton.module.scss";

interface MenuButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  /** Left icon for the menu button */
  icon: React.ReactNode;

  children: React.ReactNode;
  onClick: React.MouseEventHandler;
}

const MenuButton = ({ icon, children, onClick, ...rest }: MenuButtonProps) => {
  return (
    <button onClick={onClick} className={styles["menu-button"]} {...rest}>
      {icon}
      {children}
    </button>
  );
};

export default MenuButton;
