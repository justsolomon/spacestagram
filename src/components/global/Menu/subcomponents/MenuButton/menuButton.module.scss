@import "styles/variables";
@import "styles/mixins";

.menu-button {
  @include default-button();

  height: 50px;
  width: 100%;
  padding: 1rem 1.2rem 1rem 1rem;
  font-size: $font-size--base;

  &:hover,
  &:focus-visible {
    outline: none;
    background-color: $bg-light--primary;
  }

  svg {
    margin-right: 0.8rem;
    color: $secondary-color;
  }
}
