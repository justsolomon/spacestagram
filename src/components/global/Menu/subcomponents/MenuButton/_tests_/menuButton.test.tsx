import { render } from "@testing-library/react";
import MenuButton from "..";

describe("<MenuButton />", () => {
  it("should render the menu button correctly", () => {
    const { asFragment } = render(
      <MenuButton icon={<svg></svg>} onClick={() => {}}>
        Test menu button
      </MenuButton>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
