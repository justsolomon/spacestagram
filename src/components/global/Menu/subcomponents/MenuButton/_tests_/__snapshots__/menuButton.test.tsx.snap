// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuButton /> should render the menu button correctly 1`] = `
<DocumentFragment>
  <button
    class="menu-button"
  >
    <svg />
    Test menu button
  </button>
</DocumentFragment>
`;
